runtime/src/gpu/nvidia/gpu-nvidia.c

```diff
@@ -210,12 +215,14 @@ static void chpl_gpu_launch_kernel_help(int ln,
       // TODO this allocation needs to use `chpl_mem_alloc` with a proper desc
       kernel_params[i] = chpl_malloc(1*sizeof(CUdeviceptr));
 
-      *kernel_params[i] = chpl_gpu_mem_alloc(cur_arg_size,
-                                             CHPL_RT_MD_GPU_KERNEL_ARG,
-                                             ln, fn);
+      // TODO this now doesn't use memory allocation hooks, but maybe that's OK?
+      // TODO this doesn't work on EX, why?
+      // *kernel_params[i] = chpl_gpu_impl_mem_array_alloc(cur_arg_size, stream);
+      *kernel_params[i] = chpl_gpu_mem_alloc(cur_arg_size, CHPL_RT_MD_GPU_KERNEL_ARG,
```

Comment:
I think the follow up work here is to refactor our launchers into `chpl-gpu` layer rather than keeping them in the `impl` layer. At that point things will change quite a bit, and whatever task there is, it may be moot. I'll revise the comment here, but I think an in-code comment is sufficient for us to remember the tasks.